spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Log.java

```diff
@@ -40,7 +43,12 @@
 	 */
 	private final String event;
 
-	public Log(long timestamp, String event) {
+	@JsonCreator
+	public Log(
+			@JsonProperty(value = "timestamp", required = true) long timestamp,
```

Comment:
The problem with using reflection is that you need to make a dummy
ctor, and also have the chance for null, still. Ex. the following
still allows a later set of event to null.

@SuppressWarnings("unused") // used for reflective deserialization
private Log() {
 this.timestamp = 0L;
 this.event = "";
}